src/app-check/app-check.ts

```diff
@@ -75,17 +78,41 @@ export class AppCheck {
    * rejected.
    *
    * @param appCheckToken - The App Check token to verify.
+   * @param options - Optional {@link VerifyAppCheckTokenOptions} object when verifying an App Check Token.
    *
    * @returns A promise fulfilled with the token's decoded claims
    *   if the App Check token is valid; otherwise, a rejected promise.
    */
-  public verifyToken(appCheckToken: string): Promise<VerifyAppCheckTokenResponse> {
+  public verifyToken(appCheckToken: string, options?: VerifyAppCheckTokenOptions)
+    : Promise<VerifyAppCheckTokenResponse> {
+    this.validateVerifyAppCheckTokenOptions(options);
     return this.appCheckTokenVerifier.verifyToken(appCheckToken)
       .then((decodedToken) => {
+        if (options?.consume) {
```

Comment:
If the initial verifyToken(...) fails due to an invalid token, do we skip this step? I don't know TS that well -- if we can save the network call and throw right away that would be ideal.

Reply:
Yep! We fail fast here. If `verifyToken` throws then the code won't continue an we will not make a network call even if `consume` is set to true. I added unit tests to cover this usecase.